#169590897 Build out entry endpoints

T2Wil opened the pull request:
#### What does this PR do?
- Add entry endpoints
#### Description of Task to be completed?
- Add viewSpecificEntry controller and db queries
- Add modifyEntry controller and db queries
- Add deleteEntry controller and db queries
- Implement some tests on the endpoints
#### How should this be manually tested?
* have Postman
* clone the repository
* cd into directory
* run ``npm run start:dev``
* Get specific entry:  GET `/api/v2/entries/:entryId`
* Modify an entry :  PATCH `/api/v2/entries/:entryId`
* Delete an entry: DELETE `/api/v2/entries/:entryId`

#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
#169590897
#### Screenshots (if appropriate)
                                 `Registered users and entries in database`
![image](https://user-images.githubusercontent.com/30593186/68288828-cb55ad80-008d-11ea-96fd-5852c4c2211a.png)

![image](https://user-images.githubusercontent.com/30593186/68288883-e1fc0480-008d-11ea-8caf-e7b9f68d6728.png)

                                  `GET SPECIFIC ENTRY `
![image](https://user-images.githubusercontent.com/30593186/68288997-1079df80-008e-11ea-8fad-6a0417acb122.png)

                                   `UPDATE SPECIFIC ENTRY`
![image](https://user-images.githubusercontent.com/30593186/68290170-29839000-0090-11ea-820f-1a6505036076.png)


                                     `UPDATED ENTRIES`
![image](https://user-images.githubusercontent.com/30593186/68289154-520a8a80-008e-11ea-9e0e-bb5310a84c29.png)

![image](https://user-images.githubusercontent.com/30593186/68289201-62bb0080-008e-11ea-9e30-ed62ab09776a.png)

                                          `DELETE SPECIFIC ENTRY`
![image](https://user-images.githubusercontent.com/30593186/68289319-91d17200-008e-11ea-920e-9e18522ebe0e.png)

                                            `UPDATED ENTRIES`
![image](https://user-images.githubusercontent.com/30593186/68289354-a0b82480-008e-11ea-8450-13e195f38662.png)

![image](https://user-images.githubusercontent.com/30593186/68289398-b0d00400-008e-11ea-9ce7-8ca0af994114.png)

T2Wil committed:
ft(db-entry-endpoints): Add entry endpoints
Add viewSpecificEntry controller and db queries
Add modifyEntry controller and db queries
Add deleteEntry controller and db queries
Implement some tests on the endpoints
[Finishes ft-db-entry-endpoints-169590897]

mcaleb808 commented on server/controllers/entriesController.js:
```diff
@@ -65,13 +65,92 @@ export const viewEntries = async (req, res) => {
+      status: res.statusCode,
+      error: err,
+    });
+  }
```
@T2Wil great work implementing this features, But this block of code is repeating itself in many areas, can you create one which is reusable for code maintainability.

T2Wil replied:
@mcaleb808 
Solved!
I have created a new file containing instructions to handle errors.

T2Wil committed:
ft(db-entry-endpoints): Add entry endpoints
Add viewSpecificEntry controller and db queries
Add modifyEntry controller and db queries
Add deleteEntry controller and db queries
Implement some tests on the endpoints
[Finishes ft-db-entry-endpoints-169590897]
